libs/providers/src/lib/language-swithcer/language.provider.tsx

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useState } from 'react';
+import { LangItem, LangItems } from '.';
+import { LanguageContext } from './language.context';
+
+export interface LanguageProviderProps {
+  children: React.ReactNode;
+  langItems: LangItems;
```

Comment:
there should be a onLangSelect here, and we should call it in selectLanguage function.